griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -43,7 +43,7 @@ class OpenAiChatPromptDriver(BasePromptDriver):
         )
     )
     model: str = field(kw_only=True)
-    tokenizer: OpenAiTokenizer = field(
+    tokenizer: BaseTokenizer = field(
```

Comment:
Is there an actual use case for when we need to support other tokenizers here?

Reply:
The use-case on #512 requires it, I think. The alternative is that we add support for arbitrary models to `OpenAiTokenizer`.

Reply:
I think this is confusing. Seems like `OpenAi*` should be focused on _OpenAI-only_ things. But we could add another more generic driver and tokenizer to support _OpenAI-like_ things.

Reply:
Agreed that it is kind of confusing, but there is a trend of using OpenAi's SDK as the SDK for non-OpenAi models. And we already partially support this functionality by allowing users to set the `base_url` field. So we could create a Driver, but it would still be tightly coupled to OpenAi.

At the very least I think we need more graceful error handling in the `OpenAiTokenizer` (maybe all Tokenizers?) so it doesn't error out when people set `model` to a non-OpenAi model.

Reply:
I see...that makes sense. I think what we should fix then is the conditional code below (e.g., `if isinstance()...` by moving relevant parts into tokenizers.

Reply:
If I'm understanding correctly, that would require us to handle the result of `_prompt_stack_to_messages(prompt_stack)` (message list) in other Tokenizers which would not be straightforward. For `OpenAiTokenizer` we must have the message list, and for all other Tokenizers we must have a string.

Another potential avenue is to allow Tokenizers to take in a Prompt Stack. But the logic for converting the Prompt Stack into a string only exits in the Prompt Drivers.